modules/mantisBidAdapter.js

```diff
@@ -0,0 +1,221 @@
+        } else if (isSendable(aval)) {
+          parts.push(akey + '=' + encodeURIComponent(aval));
+        }
+      }
+    } else if (isObject(val) && val != data) {
+      jsonToQuery(val, queryKey, parts);
+    } else if (isSendable(val)) {
+      parts.push(queryKey + '=' + encodeURIComponent(val));
+    }
+  }
+  return parts.join('&');
+}
+
+function buildMantisUrl(path, data, domain) {
+  var params = {
+    referrer: document.referrer,
+    tz: new Date().getTimezoneOffset(),
+    buster: new Date().getTime(),
+    secure: isSecure()
+  };
+  if (!inIframe() || isAmp()) {
+    params.mobile = !isAmp() && isDesktop(true) ? 'false' : 'true';
+  }
+  if (window.mantis_uuid) {
+    params.uuid = window.mantis_uuid;
+  } else if (window.localStorage) {
+    var localUuid = window.localStorage.getItem('mantis:uuid');
+    if (localUuid) {
+      params.uuid = localUuid;
+    }
+  }
+  if (!inIframe()) {
+    try {
+      params.title = window.top.document.title;
+      params.referrer = window.top.document.referrer;
+      params.url = window.top.document.location.href;
+    } catch (ex) {
+    }
+  } else {
+    params.iframe = true;
+  }
+  if (isAmp()) {
+    params.amp = true;
+    if (!params.url && window.context.canonicalUrl) {
+      params.url = window.context.canonicalUrl;
+    }
+    if (!params.url && window.context.location) {
+      params.url = window.context.location.href;
+    }
+    if (!params.referrer && window.context.referrer) {
+      params.referrer = window.context.referrer;
+    }
+  }
+  Object.keys(data || {}).forEach(function (key) {
+    params[key] = data[key];
+  });
+  var query = jsonToQuery(params);
+  return (window.mantis_domain === undefined ? domain || 'https://mantodea.mantisadnetwork.com' : window.mantis_domain) + path + '?' + query;
+}
+
+const spec = {
+  code: 'mantis',
+  supportedMediaTypes: ['banner', 'video', 'native'],
+  isBidRequestValid: function (bid) {
+    return !!(bid.params.property && (bid.params.code || bid.params.zoneId || bid.params.zone));
+  },
+  buildRequests: function (validBidRequests) {
+    var property = null;
+    validBidRequests.some(function (bid) {
+      if (bid.params.property) {
+        property = bid.params.property;
+        return true;
+      }
+    });
+    const query = {
+      bids: validBidRequests.map(function (bid) {
+        return {
+          bidId: bid.bidId,
+          config: bid.params,
+          sizes: bid.sizes.map(function (size) {
+            return {width: size[0], height: size[1]};
+          })
+        };
+      }),
+      property: property,
+      version: 2
+    };
+    return {
+      method: 'GET',
+      url: buildMantisUrl('/prebid/display', query) + '&foo',
+      data: ''
+    };
+  },
+  interpretResponse: function (serverResponse) {
+    storeUuid(serverResponse.uuid);
+    return serverResponse.body.ads.map(function (ad) {
+      return {
+        requestId: ad.bid,
+        bidderCode: spec.code,
```

Comment:
Please drop this line. We add the code automatically from the spec, and it breaks aliasing if this is here.

Reply:
@mkendall07 done